src/passes/PreJSRewriter.cpp

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2021 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <pass.h>
+#include <wasm.h>
+
+#include "wasm-builder.h"
+#include <ir/literal-utils.h>
+#include <ir/localize.h>
+
+namespace wasm {
+
+struct PreJSRewriterPass : public WalkerPass<PostWalker<PreJSRewriterPass>> {
+  std::unique_ptr<Builder> builder;
```

Comment:
Rather than have a `unique_ptr<Builder>` as a class member, it would be slightly simpler just to construct a new `Builder` as a local variable in any method where it is needed.